File: leetcode-submissions/2924_Find_Champion_II.cpp
class Solution {
public:
    int findChampion(int n, vector<vector<int>>& edges) {
        // we need to store the number of incomming edges so we can easily identify which is the champion
       set<int>incomming;
        for(int i=0;i<edges.size();i++){
           incomming.insert(edges[i][1]);
        }
        //after this we will need to check the count of the missing nodes 
        int count =0;
            int ans=0;
        for(int i=0;i<edges.size();i++){
        if(incomming.find(edges[i][0])==incomming.end()){
            count ++;
            ans=edges[i][0];

        }        
        }
        return count >1?-1:ans;
    }
};

File: leetcode-submissions/2924_Find_Champion_II.ts
function findChampion(n: number, edges: number[][]): number {
    // we need to create a indegree array to store the indegree of the nodes 
    const indegree:number[]=Array.from({length:n},()=>0);
    for(const edge of edges ){
        indegree[edge[1]]++;

    }
    let count :number =0;
    let ans:number =0;
    for(let i=0;i<n;i++){
        if(indegree[i]==0){
            count ++;
            ans=i;
        }
    }
    return count >1?-1:ans;
};